packages/InlineEditing/src/Table.js

```diff
@@ -0,0 +1,63 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Table from "@paprika/table";
+import * as sc from "./Table.styles";
+
+function Editable({ children }) {
```

Comment:
If you were to export this "uncontrolled" `<Editor>` component, then consumers could use it directly with a Paprika `<Table>`, right?  

The only other thing I think I see happening is some control over the `width` and `height`?

Reply:
to have it work smoothly in the Table and don't change the API at all. 

inline-editing components have to be aware of which cell are they working on, so they can call `onSubmit({ rowIndex, columnIndex})` with the proper values and manage focus. To make this work there is a HOC that wraps the cell prop and turns the anonymous function into a component that is extended with the proper values so the end-user can have an inline-editing on a Table component without refactoring anything those props later are passdown to the inline-editing (listbox/textarea/input, etc). 

https://github.com/acl-services/paprika/pull/939/files#diff-5fb89b8beb3edab9a41d813fef3b5a5e8ba0b3344ab601ea8e90f35fa5174bdaR33-R46

**^** this is the trick `const { cell: Component, width } = child.props;`

look at this how the Listbox is aware when the onSubmit event happens and knowns which row and column executed the action.

https://github.com/acl-services/paprika/pull/939/files#diff-1a10891d491f56273522fe6251eccf716bb629606bbcdf5f3f653d7213ad62b6R22-R30

There is a lot of focus control happening as well, you can indeed just put the component and have inline editing on the table but that only will work with a click. 

Similar to the example that explains explains how it works. http://storybooks.highbond-s3.com/paprika/UXD-781-Inline-component/?path=/story/forms-inlineediting-docs--page#how-does-it-works